src/main/java/nl/tudelft/ewi/devhub/server/web/resources/ProjectAssignmentsResource.java

```diff
@@ -0,0 +1,312 @@
+
+        Assignment assignment = assignments.find(group.getCourse(), assignmentId);
+        Repository repository = gitClient.repositories().retrieve(group.getRepositoryName());
+
+        Map<String, Object> parameters = Maps.newLinkedHashMap();
+        parameters.put("user", currentUser);
+        parameters.put("group", group);
+        parameters.put("course", group.getCourse());
+        parameters.put("repository", repository);
+        parameters.put("assignment", assignment);
+        parameters.put("myDeliveries", deliveries.getDeliveries(assignment, group));
+        parameters.put("states", new ProjectResource.CommitChecker(group, buildResults));
+        parameters.put("recentCommits", repository.retrieveBranch("master").retrieveCommits(0, 25).getCommits());
+
+        List<Locale> locales = Collections.list(request.getLocales());
+        return display(templateEngine.process("courses/assignments/group-assignment-view.ftl", locales, parameters));
+    }
+
+    /**
+     * Submit an assignment for a course
+     * @param request the current HttpServletRequest
+     * @param assignmentId assignmentId for the assignment
+     * @param formData submit data
+     * @return a redirect request to the assignment page
+     * @throws IOException if an I/O error occurs
+     * @throws ApiError if an ApiError occurs
+     */
+    @POST
+    @Path("{assignmentId : \\d+}")
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    public Response postAssignment(@Context HttpServletRequest request,
+                                   @PathParam("assignmentId") Long assignmentId,
+                                   MultipartFormDataInput formData) throws IOException, ApiError, GitClientException {
+
+        Map<String, List<InputPart>> formDataMap = formData.getFormDataMap();
+        String commitId = extractString(formDataMap, "commit-id");
+        String notes = extractString(formDataMap, "notes");
+
+        Assignment assignment = assignments.find(group.getCourse(), assignmentId);
+        Delivery delivery = new Delivery();
+        delivery.setAssignment(assignment);
+        delivery.setCommitId(commitId);
+        delivery.setNotes(notes);
+        delivery.setGroup(group);
+        deliveriesBackend.deliver(delivery);
+
+        addAttachments(formDataMap, delivery);
+
+        if(!Strings.isNullOrEmpty(commitId)) {
+            tagAssignmentDelivery(commitId, assignment, delivery);
+        }
+
+        return redirect(request.getRequestURI());
+    }
+
+    private void addAttachments(Map<String, List<InputPart>> formDataMap, Delivery delivery) throws IOException, ApiError {
+        List<InputPart> attachments = formDataMap.get("file-attachment");
+        for(InputPart attachment : attachments) {
+            String fileName = extractFilename(attachment);
+            if(fileName.isEmpty()) continue;
+            InputStream in = attachment.getBody(new GenericType<InputStream>() {});
```

Comment:
There currently isn't. It would be a good idea because most probably these attachments get loaded into the memory, and students will abuse this :wink:. However, what should be the limit? For example the InCode results where 100 MB. But I would prefer a limit of +/- 20 MB.

Link to keep here: http://stackoverflow.com/a/20883976/2104280